tests/internals/cloud_events/cloud_events_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build e2e
+// +build e2e
+
+package trigger_update_so_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+const (
+	testName = "cloudevent-test"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+var (
+	namespace                  = fmt.Sprintf("%s-ns", testName)
+	scaledObjectName           = fmt.Sprintf("%s-so", testName)
+	clientName                 = fmt.Sprintf("%s-client", testName)
+	cloudEventName             = fmt.Sprintf("%s-ce", testName)
+	cloudEventHTTPReceiverName = fmt.Sprintf("%s-cloudevent-http-receiver", testName)
+	cloudEventHTTPServiceName  = fmt.Sprintf("%s-cloudevent-http-service", testName)
+	cloudEventHTTPServiceURL   = fmt.Sprintf("http://%s.%s.svc.cluster.local:8899", cloudEventHTTPServiceName, namespace)
+)
+
+type templateData struct {
+	TestNamespace              string
+	ScaledObject               string
+	ClientName                 string
+	CloudEventName             string
+	CloudEventHTTPReceiverName string
+	CloudEventHTTPServiceName  string
+	CloudEventHTTPServiceURL   string
+}
+
+const (
+	cloudEventTemplate = `
+  apiVersion: keda.sh/v1alpha1
+  kind: CloudEvent
+  metadata:
+    name: {{.CloudEventName}}
+    namespace: {{.TestNamespace}}
+  spec:
+    clusterName: cluster-sample
+    eventHandlers:
+    - type: cloud-event-http
+      name: cloud-event-http-sample
+      metadata:
+        endPoint: {{.CloudEventHTTPServiceURL}}
```

Comment:
If this is our CRD, it looks like it's not really according to the specs on https://github.com/kedacore/keda/issues/3533, no?

Reply:
No. Considering there will be multiple handlers, I changed specific handler spec to an array.

Reply:
I think we can discuss that on #3533 but my main point is there rest of the CRD which is not aligned with the requirements.

From #3533
```
apiVersion: events.keda.sh/v1alpha1
kind: EventSource # Or ClusterEventSource
metadata:
  name: operations-cross-cluster-events
spec:
  destination:
    # Support regular webhook endpoints over HTTP(S)
    http:
      uri: http://foo.bar
      authentication:
        apiKey:
          headerName: x-api-key
          valueFrom:
            secretKeyRef:
            name: secrets-operations-events
            key: webhook-api-key
  eventSubscription:
    includedEventTypes:
      - keda.example.Event
    excludedEventTypes:
      - keda.example.Event
```